packages/peregrine/lib/List/item.js

```diff
@@ -1,54 +1,63 @@
-import React, { Component } from 'react';
+import React, { useCallback } from 'react';
 import PropTypes from 'prop-types';
 
 import fromRenderProp from '../util/fromRenderProp';
 
-class Item extends Component {
-    static propTypes = {
-        classes: PropTypes.shape({
-            root: PropTypes.string
-        }),
-        hasFocus: PropTypes.bool,
-        isSelected: PropTypes.bool,
-        item: PropTypes.any.isRequired,
-        itemIndex: PropTypes.number.isRequired,
-        render: PropTypes.oneOfType([PropTypes.func, PropTypes.string])
-            .isRequired
+const isString = item => typeof item === 'string';
+
+const getChild = item => (isString(item) ? item : null);
+
+const Item = props => {
+    const {
+        uniqueID: key,
+        classes,
+        hasFocus,
+        isSelected,
+        item,
+        itemIndex,
+        render,
+        updateSelection,
+        setFocus,
+        ...restProps
+    } = props;
+    const onClick = useCallback(() => updateSelection(key), [
+        updateSelection,
+        key
+    ]);
+    const onFocus = useCallback(() => setFocus(key), [setFocus, key]);
+    const customProps = {
```

Comment:
I think you'll want to add `uniqueId` to this object so it doesn't get applied as a DOM attribute.

Reply:
In this case, it wouldn't because the execution stops at line [19 of fromRenderProp](https://github.com/magento-research/pwa-studio/blob/develop/packages/peregrine/lib/util/fromRenderProp.js#L19) and not enter [36 of fromRenderProps](https://github.com/magento-research/pwa-studio/blob/develop/packages/peregrine/lib/util/fromRenderProp.js#L36) where the filtering actually happens. This is because the element to render is a composite element.